ts/parts/Series.ts

```diff
@@ -6707,6 +6707,11 @@ H.Series = seriesType<Highcharts.LineSeries>(
             // overwrite the animation.complete option which should be available
             // to the user).
             if (!hasRendered) {
+                // Additional time if defer is defined before afterAnimate
+                // will be triggered
+                if (animDuration && defined(animObject(options.animation).defer)) {
```

Comment:
Just like `animObject` always returns an object with a duration, it should also always return a `defer` property (0 by default). That will save us from type checking in cases like this.

Reply:
@TorsteinHonsi 
I have a problem with this change and I would like to discuss it.

```
const animObject = H.animObject = function animObject(
    animation?: (boolean|Highcharts.AnimationOptionsObject)
)Highcharts.AnimationObject {
    return isObject(animation) ?
        H.merge(
            { duration: 500, defer: 0 },
            animation as Highcharts.AnimationOptionsObject
        ) as any :
        { duration: animation as boolean ? 500 : 0, defer: 0 };
}; 
```

That will be the best implementation because defer will be always assigned however requires changes in the current **Highcharts.AnimationOptionsObject** definition and probably create the new definition, like **Highcharts.AnimationObject**, which will look like this: 

```
        interface AnimationOptionsObject {
            complete?: Function;
            curAnim?: Dictionary<boolean>;
            defer: number;
            duration: number;
            easing?: (string|Function);
            step?: AnimationStepCallbackFunction;
        }
```

This implementation will avoid getting the defer or duration undefined. 

What do you think? What is your opinion about it?

Reply:
This is the only thing which left to implement (I think). Any opinion is welcome.

Reply:
Yes, defer and duration will now always be a number in the object that `animObject` returns. As opposed to the `AnimationOptionsObject` where those are optional.

One solution would be using two interfaces like you suggest.

The other solution I think would be that the _option_ is a partial of the object. @bre1470 do you think that solution would apply in this case?

Reply:
@TorsteinHonsi 
I fear that this would lead to complications later on, as both interfaces are not logically connected, just by the current implementation. I would suggest something shorter like
```ts
interface AnimationObject extends AnimationOptionsObject {
    defer: number;
    duration: number;
}
```
That connects output with input in the way, the implementation is working.

Regarding the implementation, the following line can be reduced by ` as boolean` as it provides no benefit:
```ts
{ duration: animation as boolean ? 500 : 0, defer: 0 };
```